managed/CounterStrikeSharp.API/Core/GlobalContext.cs

```diff
@@ -124,27 +130,149 @@ public void LoadAll()
 
             foreach (var path in filePaths)
             {
-                var plugin = new PluginContext(path);
-
+                if (path.Contains("disabled")) continue;
                 try
                 {
-                    _loadedPlugins.Add(plugin);
-                    plugin.Load();
+                    LoadPlugin(path);
                 }
                 catch (Exception e)
                 {
-                    _loadedPlugins.Remove(plugin);
                     Console.WriteLine($"Failed to load plugin {path} with error {e}");
                 }
             }
 
-            Console.WriteLine($"All managed modules were loaded. {_loadedPlugins.Count} plugins loaded.");
+            return _loadedPlugins.Count;
+        }
+
+        public void UnloadAllPlugins()
+        {
+            foreach (var plugin in _loadedPlugins)
+            {
+                plugin.Unload();
+                _loadedPlugins.Remove(plugin);
+            }
         }
 
         public PluginContext FindPluginByType(Type moduleClass)
         {
             var plugin = _loadedPlugins.FirstOrDefault(x => x.PluginType == moduleClass);
             return plugin;
         }
+
+        public void OnCSSCommand(CCSPlayerController? caller, CommandInfo info)
+        {
+            Utilities.ReplyToCommand(caller, "  CounterStrikeSharp was created and is maintained by Michael \"roflmuffin\" Wilson.\n" +
+                "  Counter-Strike Sharp uses code borrowed from SourceMod, Source.Python, FiveM, Saul Rennison and CS2Fixes.\n" +
+                "  See ACKNOWLEDGEMENTS.md for more information.", true);
+            return;
+        }
+
+        public void OnCSSPluginCommand(CCSPlayerController? caller, CommandInfo info)
+        {
+            switch (info.GetArg(1))
+            {
+                case "list":
+                {
+                    Utilities.ReplyToCommand(caller, "List of all plugins currently loaded by CounterStrikeSharp: ", true);
+                    if (_loadedPlugins.Count() == 0)
+                    {
+                        Utilities.ReplyToCommand(caller, $"  0 plugins loaded.", true);
+                    }
+                    else
+                    {
+                        int pluginCount = 1;
+                        foreach (var plugin in _loadedPlugins)
+                        {
+                            Utilities.ReplyToCommand(caller, $"  [{pluginCount}/{_loadedPlugins.Count()}]: \"{plugin.Name}\" (\"{plugin.Version}\")", true);
+                            pluginCount++;
+                        }
+                    }
+
+                    break;
+                }
+                case "start":
+                case "load":
+                {
+                    if (info.ArgCount < 2)
+                    {
+                        Utilities.ReplyToCommand(caller, "Valid usage: css_plugins start/load [absolute plugin path] (e.g \"plugins/TestPlugin/TestPlugin.dll\")\n", true);
```

Comment:
Noting that this is actually relative path not absolute, we could also probably consider adding support for both kinds of loading.

```suggestion
                        Utilities.ReplyToCommand(caller, "Valid usage: css_plugins start/load [relative plugin path] (e.g \"plugins/TestPlugin/TestPlugin.dll\")\n", true);
```

Reply:
I think we should also do a happy path attempt here and try to load purely based on a filename, i.e if you type `TestPlugin` it will first try `plugins/TestPlugin/TestPlugin.dll`, and then failing that move onto a path based approach.

Reply:
Gave that a shot in my latest commit.